plugins/catalog-react/api-report.md

```diff
@@ -503,6 +511,13 @@ export type UnregisterEntityDialogProps = {
   entity: Entity;
 };
 
+// @public
+export function useAllKinds(): {
+  loading: boolean;
+  error?: Error;
+  allKinds: string[];
+};
```

Comment:
Sorry. I'm a bit new in Javascript/Typescript. I figured the way to handle this is specifying `@internal` on the functions instead of `@public`.

https://api-extractor.com/pages/tsdoc/tag_internal/

Ie.
```
/**
 * Fetch and return all availible kinds.
 * @internal
 */
export function useAllKinds(): {
```

But I only get:

```
# Generating package API reports
## Processing plugins/catalog-react
Error: Internal Error: Unable to analyze the export "useAllKinds" in
/home/gustaflh/git_repos/backstage/dist-types/plugins/catalog-react/src/utils/kindFilterUtils.d.ts

You have encountered a software defect. Please consider reporting the issue to the maintainers of this application.
    at ExportAnalyzer._getExportOfAstModule (backstage/node_modules/@microsoft/api-extractor/src/analyzer/ExportAnalyzer.ts:779:13)
    at ExportAnalyzer._getExportOfSpecifierAstModule (backstage/node_modules/@microsoft/api-extractor/src/analyzer/ExportAnalyzer.ts:767:39)
```

Am I missing something obvious here? (Most likely)

Reply:
No, what I mean is that they should have no tag at all, and just not being `export`-ed all the way up to the root index.ts file - that way they wouldn't be appearing in the api-report.

Reply:
Hang on. We are talking about two packages (`CatalogKindHeader -> plugin-catalog` and `EntityKindPicker -> plugin-catalog-react`. I kinda need to put the utility code somewhere. And it makes sense to chose `plugin-catalog-react`. I don't see how I can solve this without exporting the helper functions.

Reply:
Worst case duplicating them (and keeping them out of the external exports) if they are used in both, right? Anything added in the public surface gets stuck there, as a promise for the future, including semver implications of that - so I want to keep anything non-core out of there as much as possible